sdk/search/azure-search-documents/src/test/java/com/azure/search/documents/TestHelpers.java

```diff
@@ -152,4 +98,93 @@ public static MatchConditions generateIfExistsAccessCondition() {
     public static MatchConditions generateIfNotChangedAccessCondition(String eTag) {
         return new MatchConditions().setIfMatch(eTag);
     }
+
+    /**
+     * Assert whether two objects are equal.
+     *
+     * @param expected The expected object.
+     * @param actual The actual object.
+     */
+    public static void assertObjectEquals(Object expected, Object actual) {
+        JacksonAdapter jacksonAdapter = new JacksonAdapter();
+        try {
+            assertEquals(jacksonAdapter.serialize(expected, SerializerEncoding.JSON),
+                jacksonAdapter.serialize(actual, SerializerEncoding.JSON));
+        } catch (IOException ex) {
+            fail("There is something wrong happen in serializer.");
+        }
+    }
+
+    /**
+     * Assert whether two objects are equal.
+     *
+     * @param expected The expected object.
+     * @param actual The actual object.
+     * @param ignoredDefaults Set to true if it needs to ignore default value of expected object.
+     * @param ignoredFields Varargs of ignored fields.
+     */
+    public static void assertObjectEquals(Object expected, Object actual, boolean ignoredDefaults,
+        String... ignoredFields) {
+        ObjectMapper mapper = new ObjectMapper();
+        ObjectNode expectedNode = mapper.valueToTree(expected);
+        ObjectNode actualNode = mapper.valueToTree(actual);
+        assertOnMapIterator(expectedNode.fields(), actualNode, ignoredDefaults,ignoredFields);
+    }
+
+    private static void assertOnMapIterator(Iterator<Map.Entry<String, JsonNode>> expectedNode,
+        ObjectNode actualNode, boolean ignoredDefaults, String[] ignoredFields) {
+        while (expectedNode.hasNext()) {
+            assertTrue(actualNode.fields().hasNext());
+            Map.Entry<String, JsonNode> expectedField = expectedNode.next();
+            String fieldName = expectedField.getKey();
+            Set<String> ignoredFieldSet = new HashSet<>(Arrays.asList(ignoredFields));
```

Comment:
You'll want to move this outside of the while loop, right now this will be recreated each time.